### ISSUE TITLE
Current COMPADRE release and development versions fail to meet CompadreData/CompadreM definition

### ISSUE THREAD

**Issue by levisc8** [jonesor/Rcompadre#16]
Currently, Compadre v4.0.1 and Compadre vX.X.X (unreleased version) fail to meet the CompadreData class definition. It seems that 24 of the A and/or U matrices have negative elements which are currently rejected during validation of `CompadreM` class objects. For example: 

`load('COMPADRE_V.4.0.1.RData')`
`Compadre <- asCompadreData(compadre)`
`Error in validObject(.Object) : 
  invalid class “CompadreM” object: matU is not nonnegative`

I've extracted the DOIs for the offending studies and have found the papers. 

In the Jongejans 2010 paper (DOI: 10.1111/j.1365-2745.2009.01612.x, _Carlina vulgaris_, matrix b2), there was no data entry error - the element as calculated by them is slightly negative. 

In the Kesler et al 2008 paper (DOI: 10.1007/s00442-008-1022-1, 23 U matrices with slightly negative elements), I am guessing the negative matrix elements are a byproduct of the regressions they used. However, as noted in the Observation column of the metadata, the matrices were provided by the author so I can't check them by hand at the moment.

I remember discussing this issue in Ghent and seem to recall that we wanted to retain the matrices as reported by the authors. I think this would mean we need to loosen our definition of the class (perhaps throw a warning instead of an error) or subset the problem studies out each time a user submits a query that includes them (which would largely defeat the purpose of retaining them in the first place). Alternative solutions that don't require loosening the class definition or getting rid of the offending studies would be ideal, but I can't think of any at the moment.

@jonesor @robcito @tdjames1 @iainmstott  - thoughts on how to correct this? 

**tldr**: 2 studies with a total of 24 matrices in Compadre have negative matrix elements and prevent it from satisfying our current class definition. Relax the class definition, get rid of the studies, clarify with authors, or something else entirely?

DOIs for the studies in question: `10.1111/j.1365-2745.2009.01612.x` and `10.1007/s00442-008-1022-1`

**Comment by iainmstott** [comment]
That is... frustrating. I didn't realise there would be such matrices in the data. I can sort of see why in exceptional cases like these it might be necessary to relax the rules of the class, though. What I would suggest doing instead is that we migrate this check to functions in  `Rage` which assume a nonnegative matrix so that whilst the data may be included in com(p)adre and pass the checks of the ´class, that they can't be in calculations whose assumptions they violate (this is the approach I've taken in `popdemo`)

**Comment by iainmstott** [comment]
I can take care of coding this in the `CompadreM` class file. I've assigned myself the task

**Comment by iainmstott** [comment]
I've removed the need for matrices to be nonnegative... hopefully now the current databases will work as CompadreM objects @levisc8 

We'll need to be careful about making sure to check for nonnegativity in other functions in Rcompadre, Rage and popdemo. For popdemo I've been thinking about writing a checking function that runs through common culprits which can then be run at the start of each functions with arguments to select or deselect the components to check against. Maybe this would be a good idea for Rcompadre and Rage also...

**Comment by levisc8** [comment]
@iainmstott I've tested the new version and it seems to work. I definitely agree that we need to be mindful of implementing those checks in `Rage` and `popdemo` though.
